xlators/storage/posix/src/posix-gfid-path.c

```diff
@@ -165,6 +165,18 @@ posix_get_gfid2path(xlator_t *this, inode_t *inode, const char *real_path,
             /* Convert pargfid to path */
             ret = posix_resolve_dirgfid_to_path(pargfid, priv->base_path,
                                                 &xattr_value[37], &paths[i]);
+            if (ret == -1) {
+                *op_errno = errno;
+                gf_msg(this->name, GF_LOG_ERROR, errno, P_MSG_PGFID_OP,
+                       "Parent GFID to path conversion failed"
+                       " %s: key = %s parebt GFID: %s base name: %s",
+                       real_path, keybuffer, pargfid_str, &xattr_value[37]);
+                /* if ret == -1, then paths[i] will not be allocated, hence
+                 * no need to free the memory here
+                 */
+                break;
```

Comment:
Shouldn't we jump to `err` label instead of continuing with incomplete information ?

Reply:
@xhernandez  On line number 148 on the same file, we are doing a break on a similar error case and continuing gfid2path conversion if at least one such key is available. I guess it is for files with multiple hardlink on different parent gfid(I have not looked at it more deeply). I first thought error out, Since we were continuing with at least one key, I took the same approach here as well. I think we can change it if you think it is better to fail the call.

Reply:
You are right. I think it's fine to try the remaining paths if there's an error.